src/hooks/useFilterInputChangeHandler.ts

```diff
@@ -0,0 +1,52 @@
+import { useState } from "react";
+import debounce from "lodash.debounce";
+import queryString from "query-string";
+
+const arrayFormat = "comma";
+
+const updateQueryParam = debounce(
+  (
+    urlSearchParam: string,
+    inputValue: string,
+    search: string,
+    replace: (path: string) => void,
+    pathname: string
+  ) => {
+    const nextQueryParams = queryString.stringify({
+      ...queryString.parse(search, { arrayFormat }),
+      [urlSearchParam]: inputValue === "" ? undefined : inputValue
+    });
+    replace(`${pathname}?${nextQueryParams}`);
+  },
+  250
+);
+
+type InputEvent = React.ChangeEvent<HTMLInputElement>;
+
+/**
+ * useFilterInputChangeHandler updates the url query param (urlSearchParam) by the
+ * the value user enters into the input field.
+ * @param  {string} urlSearchParam the url search param that should update
+ * @param  {string} pathname url pathname
+ * @param  {string} search url search
```

Comment:
yeah true. the hook is going to be twice for the tasks so i figured it would be more performant to pass location as a parameter. but now that you mention it i don't think it will be a problem to call it in the hook.